internal/services/costmanagement/view_resource_base.go

```diff
@@ -0,0 +1,586 @@
+
+								"sorting": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"direction": {
+												Type:         pluginsdk.TypeString,
+												Required:     true,
+												ValidateFunc: validation.StringInSlice(views.PossibleValuesForReportConfigSortingType(), false),
+											},
+											"name": {
+												Type:     pluginsdk.TypeString,
+												Required: true,
+											},
+										},
+									},
+								},
+
+								"grouping": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"type": {
+												Type:     pluginsdk.TypeString,
+												Required: true,
+											},
+											"name": {
+												Type:     pluginsdk.TypeString,
+												Required: true,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"kpi": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+					"type": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(views.PossibleValuesForKpiTypeType(), false),
+					},
+				},
+			},
+		},
+
+		"pivot": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+					"type": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(views.PossibleValuesForPivotTypeType(), false),
+					},
+				},
+			},
+		},
+	}
+
+	for k, v := range fields {
+		output[k] = v
+	}
+
+	return output
+}
+
+func (br costManagementViewBaseResource) attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (br costManagementViewBaseResource) createFunc(resourceName, scopeFieldName string) sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.CostManagement.ViewsClient
+			id := views.NewScopedViewID(metadata.ResourceData.Get(scopeFieldName).(string), metadata.ResourceData.Get("name").(string))
+
+			existing, err := client.GetByScope(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return tf.ImportAsExistsError(resourceName, id.ID())
+			}
+
+			if err := createOrUpdateCostManagementView(ctx, client, metadata, id, nil); err != nil {
```

Comment:
We will need to split this function and inline it. For context we will need to split all shared create and update methods and use `d.HasChanges` in the update so that `ignore_changes` will work.

Reply:
Will do, thanks :)

Reply:
@stephybun Can you elaborate a bit how this works?

- Should I check the changes for every field?
- Should I only send the fields which have changed or first Get the whole resource and update the changed fields?

Reply:
I've enhanced it on highest level, so not every field is checked before expanding but the main blocks are.